### PR TITLE
:bug: fix(main): Add SystemExit exception

### LINKED ISSUE #50: python -m rotkehlchen --help throws an exception
Trying to invoke `--help` on the python backend shows the usage and help text but after that throws an exception.

This is the output:
```
  --loglevel {debug,info,warn,error,critical}
                        Choose the logging level
  --logfromothermodules
                        If given then logs from all all imported modules that
                        use the logging system will also be visible.
CRITICAL:root:Traceback (most recent call last):
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/__main__.py", line 12, in main
    rotkehlchen_server = RotkehlchenServer()
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/server.py", line 65, in __init__
    self.args = app_args()
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/args.py", line 72, in app_args
    args = p.parse_args()
  File "/usr/lib64/python3.6/argparse.py", line 1730, in parse_args
    args, argv = self.parse_known_args(args, namespace)
  File "/usr/lib64/python3.6/argparse.py", line 1762, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/usr/lib64/python3.6/argparse.py", line 1968, in _parse_known_args
    start_index = consume_optional(start_index)
  File "/usr/lib64/python3.6/argparse.py", line 1908, in consume_optional
    take_action(action, args, option_string)
  File "/usr/lib64/python3.6/argparse.py", line 1836, in take_action
    action(self, namespace, argument_values, option_string)
  File "/usr/lib64/python3.6/argparse.py", line 1021, in __call__
    parser.exit()
  File "/usr/lib64/python3.6/argparse.py", line 2376, in exit
    _sys.exit(status)
SystemExit: 0

Failed to start rotkehlchen backend:
Traceback (most recent call last):
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/__main__.py", line 12, in main
    rotkehlchen_server = RotkehlchenServer()
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/server.py", line 65, in __init__
    self.args = app_args()
  File "/home/lefteris/w/rotkehlchen/rotkehlchen/args.py", line 72, in app_args
    args = p.parse_args()
  File "/usr/lib64/python3.6/argparse.py", line 1730, in parse_args
    args, argv = self.parse_known_args(args, namespace)
  File "/usr/lib64/python3.6/argparse.py", line 1762, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/usr/lib64/python3.6/argparse.py", line 1968, in _parse_known_args
    start_index = consume_optional(start_index)
  File "/usr/lib64/python3.6/argparse.py", line 1908, in consume_optional
    take_action(action, args, option_string)
  File "/usr/lib64/python3.6/argparse.py", line 1836, in take_action
    action(self, namespace, argument_values, option_string)
  File "/usr/lib64/python3.6/argparse.py", line 1021, in __call__
    parser.exit()
  File "/usr/lib64/python3.6/argparse.py", line 2376, in exit
    _sys.exit(status)
SystemExit: 0
```

This should be really easy to fix. Simply add another `except` [here](https://github.com/rotkehlchenio/rotkehlchen/blob/486866234804c64ee7e7c581cc1bc7f7841bec96/rotkehlchen/__main__.py#L13) and catch the `SystemExit` exception and in the case it's a clean exit, let it exit cleanly and if not print a stack trace.

Like that:

```
    except SystemExit as e:
        if e.code is None or e.code == 0:
            sys.exit(0)
        else:
            tb = traceback.format_exc()
            logging.critical(tb)
            print("Failed to start rotkehlchen backend:\n{}".format(tb))
            sys.exit(e.code)
```

```
```

### PR DESCRIPTION
**Pull request by anselal**
Closes #50 

I hope I did it as you expected

### CODE REVIEW

**Comment by LefterisJP** [comment]
Hello! Thank you for your contribution. It's close to what we need but needs one small change.

File: rotkehlchen/__main__.py [hunk 1]
```diff
@@ -10,11 +10,14 @@ def main():
     from rotkehlchen.server import RotkehlchenServer
     try:
         rotkehlchen_server = RotkehlchenServer()
-    except: # noqa
-        tb = traceback.format_exc()
-        logging.critical(tb)
-        print("Failed to start rotkehlchen backend:\n{}".format(tb))
-        sys.exit(1)
+    except SystemExit as e:
```
Comment (LefterisJP) [inline rotkehlchen/__main__.py]: It's duplicating code but you would need to also leave the previous except there.

So final result should be like:

```python
except SystemExit as e:
    if e.code is None or e.code == 0:
        sys.exit(0)
    else:
        tb = traceback.format_exc()
        logging.critical(tb)
        print("Failed to start rotkehlchen backend:\n{}".format(tb))
        sys.exit(1)
except: # noqa
    tb = traceback.format_exc()
    logging.critical(tb)
    print("Failed to start rotkehlchen backend:\n{}".format(tb))
    sys.exit(1)
```

**Comment by LefterisJP** [comment]
@anselal ping

**Comment by anselal** [comment]
@LefterisJP pong. Sorry, I was out with family. I will fix the code.
I noticed that you said to add another exception but I wanted to get rid of the duplicate code.
Isn't there a way to do it ?

**Comment by LefterisJP** [comment]
@anselal yep you could just abstract them all into one function on the top of this module but perhaps it's an overkill just for those two places.

**Comment by anselal** [comment]
If this is the only way I guess you are right, no need to abstract the code into a function.

Some tests failed for some reason but as I understand it has nothing to do with this PR

**Comment by LefterisJP** [comment]
Since it's one diff can you also squash them into one commit? 

Don't worry about the tests, there is one flaky test that I need to take a look at.

**Comment by LefterisJP** [comment]
And then add a second commit in the changelog saying that this bug was fixed? You can see the format [here](https://raw.githubusercontent.com/rotkehlchenio/rotkehlchen/master/CHANGELOG.rst)

Essentially should be added to the top as:

```rst
* :bug:`50` rotkehlchen --help now works again
```

**Comment by anselal** [comment]
Sure. Is it ok if I close this PR and open a new one instead ?

**Comment by LefterisJP** [comment]
@anselal You can do it, but there is no reason. You can rebase your commits on master and squash them all into one and then add 1 more and force push into your branch.

**Comment by anselal** [comment]
Thank you for the tip ;)

I hope it is OK now. Sorry for the mess

**Comment by LefterisJP** [comment]
Moved it to this PR: https://github.com/rotkehlchenio/rotkehlchen/pull/55

Since you did not use a specific branch but commited it into your master github closed the PR when I tried to rebase your changes into the upstream master. In the future try to use feature branches for PRs as it will also be easier and cleare rfor you to see what diffs each branch contains.

**Comment by anselal** [comment]
I always use feature branches. This was supposed to be a one commit fix, on my end, this is why I pushed directly to my master branch and this is why I ask to create another PR in the first place ;)

Next time I will try to be more careful. I need more practice with branches and merges...

Again, sorry for the mess.

**Comment by LefterisJP** [comment]
yeah no problem Just wanted to explain why I had to open another PR.

**Pull request closed without merging**